includes/ConfigFileParser.hpp

```diff
@@ -26,8 +29,9 @@ class ConfigFile : public Parser {
         virtual ~ConfigFile();
 
         /** Getters */
-        Server const & getServerData() const;
-        Location const & getLocationData() const;
+//        Server const & getServerData() const;
+//        Location const & getLocationData() const;
+        std::map<Server*, std::vector<Location> > const & getServerDataMap() const;
```

Comment:
Now we don't need to getServerData() AND getLocationData(), we just call the getServerDataMap() and we have access to all data.